lib/streamlit/string_util.py

```diff
@@ -67,6 +73,39 @@ def validate_emoji(maybe_emoji: str | None) -> str:
         )
 
 
+def validate_material_icon(maybe_material_icon: str | None) -> str:
+
+    supported_icon_packs = [
+        "material",
+    ]
+
+    if maybe_material_icon is None:
+        return ""
+
+    icon_regex = r"^\s*:(.*)\/(.*):\s*$"
+    icon_match = re.match(icon_regex, maybe_material_icon)
+
+    if not icon_match:
+        raise StreamlitAPIException(
+            f'The value `"{maybe_material_icon}"` is not a valid Material icon. '
+            f"Please use a Material icon shortcode like **`:material/thumb_up:`**"
+        )
+
+    pack_name, icon_name = icon_match.groups()
+
+    if (
+        pack_name not in supported_icon_packs
+        or not icon_name
+        or not is_material_icon(icon_name)
+    ):
+        raise StreamlitAPIException(
+            f'The value `"{maybe_material_icon}"` is not a valid Material icon.'
```

Comment:
I think the text is OK for an initial release but I agree a how-to link would be helpful to add. if we add a link I think it's preferable to use one that we control (such as a small how-to) where we can give detailed instructions vs dropping someone in a google owned page. This could also be covered in the API reference for the affected commands (or linked from there perhaps).

cc @sfc-gh-dmatthews , will defer to when / how she wants to handle it if she has a preference

Note: this would also be an argument to consolidate / generalize the error text as mentioned somewhere else so it's easier to update later